unbalanced_dataset/base_sampler.py

```diff
@@ -48,17 +50,19 @@ def __init__(self, ratio='auto', random_state=None, verbose=True):
         # over the number of samples in the majority class. Thus, the ratio
         # cannot be greater than 1.0
         if isinstance(ratio, float):
-            if ratio > 1.0 or ratio < 0.:
-                raise ValueError('Ratio cannot be greater than one or negative'
-                                 '. Otherwise the majority class become'
-                                 ' minority.')
+            if ratio > 1.0:
+                raise ValueError('Ration cannot be greater than one.')
+            elif ratio < 0.:
+                raise ValueError('Ratio cannot be negative.')
             else:
                 self.ratio_ = ratio
         elif isinstance(ratio, string_types):
             if ratio == 'auto':
                 self.ratio_ = ratio
             else:
                 raise ValueError('Unknown string for the parameter ratio.')
+        else:
```

Comment:
This would fail if a value of 1 (not 1.) is passed, however this is likely the most common scenario given that python is fairly lenient in this aspect.

On this note, what would a ratio of 0 (or 0.0) mean? Should we enforce ratio > 0, as opposed to ratio >= 0?